internal/consumer/ackMgr.go

```diff
@@ -195,3 +199,24 @@ func (l *threadSafeSortedList) Add(value int64) (list.Address, error) {
 	}
 	return addr, err
 }
+
+func (l *threadSafeSortedList) Reset() <-chan struct{} {
```

Comment:
why not make this a blocking call i.e. don't return a channel for the caller to wait, but instead have this method return only after the reset is complete. I don't see any use case for non-blocking API as of today, so, blocking would help avoid bugs where someone calls Reset and forgets about it

Reply:
Done.